TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -16,38 +16,79 @@ namespace TypeCobol.Compiler.Nodes {
     public class DataDivision: GenericNode<DataDivisionHeader>, Parent<DataSection> {
 
         public const string NODE_ID = "data-division";
+
+        // The 5 (optional) data sections
+        private FileSection _fileSection;
+        private GlobalStorageSection _globalStorageSection;
+        private WorkingStorageSection _workingStorageSection;
+        private LocalStorageSection _localStorageSection;
+        private LinkageSection _linkageSection;
+
+        public FileSection FileSection => _fileSection;
+        public GlobalStorageSection GlobalStorageSection => _globalStorageSection;
+        public WorkingStorageSection WorkingStorageSection => _workingStorageSection;
+        public LocalStorageSection LocalStorageSection => _localStorageSection;
+        public LinkageSection LinkageSection => _linkageSection;
+
         public DataDivision(DataDivisionHeader header): base(header) { }
         public override string ID { get { return NODE_ID; } }
 
         public override void Add(Node child, int index = -1) {
-            if (index <= 0) index = WhereShouldIAdd(child.GetType());
+            if (child is DataSection section) UpdateSection(section);
```

Comment:
We should look for a way to update section and find proper index in one go  to avoid multiple type checks